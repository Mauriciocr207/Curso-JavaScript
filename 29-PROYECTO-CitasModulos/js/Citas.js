export class Citas {
    static citas = [];
    
    static add(obj) {
        this.citas = [...this.citas, {
            id: Date.now(),
            ...obj,
        }];
    }

    static update(citaObj) {
        this.citas = this.citas.map((cita => cita.id === citaObj.id ? citaObj:cita));
    }

    static delete(idCita) {
        this.citas = this.citas.filter(({id}) => id !== idCita);
    }

    static getAll() {
        return this.citas;
    }

    static getById(id) {
        return this.citas.find(cita => cita.id === parseInt(id));
    }

    static validate(cita) {
        const {
            mascota, 
            propietario, 
            telefono, 
            fecha,
            hora,
            sintomas
        } = cita;
        if(mascota === "" || propietario  === "" || telefono  === "" || fecha  === "" || hora  === "" || sintomas === "") {
            return false;
        }
        return true;
    }
    
}