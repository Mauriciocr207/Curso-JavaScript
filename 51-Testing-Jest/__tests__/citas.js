import Citas from "../js/classes/Citas";

describe('probar clase citas', () => {
    const citas = new Citas();
    const citaObj = {
        id: Date.now(),
        mascota: 'Hook',
        propietario: 'Juan',
        telefono: '1232321',
        fecha: '10-12-2020',
        hora:'10:20',
        sintomas: 'Solo duerme'
    }

    test('addCita', () => {
        citas.agregarCita(citaObj);
        expect(citas).toMatchSnapshot();
    })
    
    test('editCita', () => {
        const citaActualizada = citaObj;
        citaActualizada.mascota = "Nuevo nombre de mascota";
        citas.editarCita(citaActualizada);
        expect(citas).toMatchSnapshot();
    })
    
    test('deleteCita', () => {
        citas.eliminarCita(citaObj.id);
        expect(citas).toMatchSnapshot();
    })
})