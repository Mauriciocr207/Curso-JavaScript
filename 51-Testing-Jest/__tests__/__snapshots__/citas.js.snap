// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probar clase citas addCita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2020",
      "hora": "10:20",
      "id": 1711435048754,
      "mascota": "Hook",
      "propietario": "Juan",
      "sintomas": "Solo duerme",
      "telefono": "1232321",
    },
  ],
}
`;

exports[`probar clase citas deleteCita 1`] = `
Citas {
  "citas": [],
}
`;

exports[`probar clase citas editCita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2020",
      "hora": "10:20",
      "id": 1711435048754,
      "mascota": "Nuevo nombre de mascota",
      "propietario": "Juan",
      "sintomas": "Solo duerme",
      "telefono": "1232321",
    },
  ],
}
`;
